lib/swoc/include/swoc/IntrusiveHashMap.h

```diff
@@ -514,6 +514,15 @@ IntrusiveHashMap<H>::insert(value_type *v) {
     if (spot != bucket->_v) {
       mixed_p = true; // found some other key, it's going to be mixed.
     }
+    if (spot != limit) {
+      // If an equal key was found, walk past those to insert at the upper end of the range.
+      do {
+        spot = H::next_ptr(spot);
+      } while (spot != limit && H::equal(key, H::key_of(spot)));
+      if (spot != limit) { // something not equal past last equivalent, it's going to be mixed.
+        mixed_p = true;
+      }
+    }
```

Comment:
This puts back what we reverted because origin side session pooling was broken by it. See #10992 where we pulled this out.

To address the functional concerns, see my patch below in HttpSessionManager.cc